scikeras/wrappers.py

```diff
@@ -759,7 +769,7 @@ def score(self, X, y, sample_weight=None):
 
         return self.scorer(y, y_pred, sample_weight=sample_weight, **score_args)
 
-    def get_meta(self) -> Dict[str, Any]:
+    def get_metadata(self) -> Dict[str, Any]:
```

Comment:
Why does this function loop over `self.__dict__` and return all attributes that satisfy these conditions?

1. Not one of `_meta`, `_fit_kwargs`, `_compile_kwargs`, etc.
2. Not a parameter (epochs, model, etc)
3. Ending in `_` or starting with `_` (which includes private attributes!).

I think this function should raise a `NotFittedError` if not fitted and return `self.meta_` otherwise as above.

Reply:
(I think this function should be deleted; it's not relevant unless `fit` called, so I think it should be property, `self.metadata_`).

Reply:
How about this:

```python3
@property
def metadata_(self) -> Dict[str, Any]:
    """Meta parameters (parameters created by fit, like
    n_features_in_ or target_type_).

    Returns
    -------
    Dict[str, Any]
        Dictionary of meta parameters
    """
    if not hasattr(self, "_random_state"):
        # Checking `_random_state` is somewhat arbitrary
        # Ideally, we would check `model_`, but this property
        # is accessed internally between before `model_` is generated
        raise NotFittedError(
            "The `metadata_` attribute cannot be accessed"
            " on an unfitted estimator. You must first call `fit`."
        )
    return {
        k: self.__dict__[k]
        for k in self._meta
        if k in self.__dict__
    }
```

Reply:
I think this resolved because:
1. The new implementation does not depend on any other attributes/sets.
2. Does not return attributes starting with `_` (i.e. not private attributes).
3. Is not public (and thus does not need to raise a `NotFittedErrror`).

Do you agree?